### ISSUE TITLE
Enable scrolling for longer filenames

### ISSUE THREAD

**Issue by ccatlett1984** [Sovol3d/SV06-PLUS#2]
The SV06 will scroll filenames that do not fit on the screen, it would be nice if the SV06 Plus would do this for the selected file prior to starting the print. Currently I have to use short names, or only put a few files on the SD card to ensure selecting the correct file to print.

**Comment by Andy3dprinter** [comment]
Yes I agree. Please enable longer file names like the SV06 does. And can you also update Marlin to the newest version? So we can have input shaping? tks

**Comment by Sovol3d** [comment]
Hello,
I am sorry for the delay in the response during the weekend off. I hope you had a nice weekend off.
We are sorry for the inconvenience caused to you. Regarding the problem you encountered, we suggest you don't use special symbol for naming, and file names should not exceed 14 characters.
Best regards,
Jan
website: sovol3d.com
Follow us to get an exclusive 3D printing information
Sovol Official User Group: https://www.facebook.com/groups/sovol3d <https://www.facebook.com/sovol3d/ >
Facebook Page:https://www.facebook.com/sovol3d <https://twitter.com/Sovol3d >
Youtube: https://www.youtube.com/channel/UCsKntuwh7dhAGep8R6kasog <https://www.youtube.com/channel/UCsKntuwh7dhAGep8R6kasog >
------------------------------------------------------------------
From:Andy3dprinter ***@***.***>
Sent At:2023 Aug. 4 (Fri.) 11:38
To:Sovol3d/SV06-PLUS ***@***.***>
Cc:Subscribed ***@***.***>
Subject:Re: [Sovol3d/SV06-PLUS] Enable scrolling for longer filenames (Issue #2)
Yes I agree. Please enable longer file names like the SV06 does. And can you also update Marlin to the newest version? So we can have input shaping? tks

**Comment by ccatlett1984** [comment]
So, the official stance is that the SV06 gets long filename support.... And
the Plus doesn't.....

**Comment by Sovol3d** [comment]
Hello,
We recommend less than 14 characters in case the printer doesn't recognize your file. If the file name is too long, the screen scrolls through the file name.
Best regards,
Jan
website: sovol3d.com
Follow us to get an exclusive 3D printing information
Sovol Official User Group: https://www.facebook.com/groups/sovol3d <https://www.facebook.com/sovol3d/ >
Facebook Page:https://www.facebook.com/sovol3d <https://twitter.com/Sovol3d >
Youtube: https://www.youtube.com/channel/UCsKntuwh7dhAGep8R6kasog <https://www.youtube.com/channel/UCsKntuwh7dhAGep8R6kasog >
------------------------------------------------------------------
From:ccatlett1984 ***@***.***>
Sent At:2023 Aug. 7 (Mon.) 21:10
To:Sovol3d/SV06-PLUS ***@***.***>
Cc:info ***@***.***>; Comment ***@***.***>
Subject:Re: [Sovol3d/SV06-PLUS] Enable scrolling for longer filenames (Issue #2)
 So, the official stance is that the SV06 gets long filename support.... And 
 the Plus doesn't.....

**Comment by smkent** [comment]
Copying [my comment from #1](https://github.com/Sovol3d/SV06-PLUS/issues/1#issuecomment-1858395698):

> [In digging around the SV06 Plus firmware to get LCD progress working for USB prints](https://github.com/smkent/sovol-sv06-plus-firmware), I've learned a bit about how the LCD works and how file names are populated. The LCD is its own little microcontroller that talks to the printer's own firmware via a serial connection; it's not like a monitor driven by a display signal from a computer. What this means is the LCD's pages are all preprogrammed and the printer firmware tells the LCD which page to go to in response to certain events, with some fields for populating data (like file names).
> 
> The SV06 Plus LCD has 4 predefined file selection pages with 5 file name slots each, and the printer firmware correspondingly fills up to these [20 file name slots](https://github.com/Sovol3d/SV06-PLUS/blob/ea9f2cb03ba9eecddcb0d17d33bc66e2e2f98873/MarlinSV06Plus/Marlin/src/lcd/sv06p/LCD_RTS.h#L11) (with [up to 20 characters each](https://github.com/Sovol3d/SV06-PLUS/blob/ea9f2cb03ba9eecddcb0d17d33bc66e2e2f98873/MarlinSV06Plus/Marlin/src/lcd/sv06p/LCD_RTS.h#L10)) using [file names from the root directory of the SD card](https://github.com/Sovol3d/SV06-PLUS/blob/ea9f2cb03ba9eecddcb0d17d33bc66e2e2f98873/MarlinSV06Plus/Marlin/src/lcd/sv06p/LCD_RTS.cpp#L134-L157).

I don't have an SV06, but my (possibly incorrect) understanding of the SV06 LCD and other simple text LCDs is the firmware sends the LCD the full text strings (e.g. long file names), and the LCD performs the scrolling. The SV06 Plus' LCD as programmed has fields that can be populated (e.g. file names), but those are displayed statically.

It might be possible to implement long file name scrolling in the SV06 Plus firmware. The firmware's `LCD_RTS.cpp` would need new logic to keep track of whether the current LCD screen is a file name display, and would need to update the LCD's file name fields for the current page every tick with a different string to simulate scrolling (e.g. tick 1 `"file..."`, tick 2 `"...ile_n..."`, tick 3 `"..._name"`).
